app/src/main/java/dev/arkbuilders/arkmemo/ui/views/SwitchButton.kt

```diff
@@ -0,0 +1,875 @@
+            if (valueAnimator.isRunning) {
+                valueAnimator.cancel()
+            }
+            animateState = ANIMATE_STATE_PENDING_RESET
+            beforeState.copy(viewState)
+            if (isChecked()) {
+                setCheckedViewState(afterState)
+            } else {
+                setUncheckViewState(afterState)
+            }
+            valueAnimator.start()
+        }
+    }
+
+    private fun pendingSettleState() {
+        if (valueAnimator.isRunning) {
+            valueAnimator.cancel()
+        }
+        animateState = ANIMATE_STATE_PENDING_SETTLE
+        beforeState.copy(viewState)
+        if (isChecked()) {
+            setCheckedViewState(afterState)
+        } else {
+            setUncheckViewState(afterState)
+        }
+        valueAnimator.start()
+    }
+
+    override fun setOnClickListener(l: OnClickListener?) {}
+    override fun setOnLongClickListener(l: OnLongClickListener?) {}
+    fun setOnCheckedChangeListener(l: OnCheckedChangeListener?) {
+        onCheckedChangeListener = l
+    }
+
+    interface OnCheckedChangeListener {
+        fun onCheckedChanged(view: SwitchButton?, isChecked: Boolean)
+    }
+
+    private var shadowRadius = 0
+    private var shadowOffset = 0
+    private var shadowColor = 0
+    private var viewRadius = 0f
+    private var buttonRadius = 0f
+    private var height = 0f
+    private var width = 0f
+    private var left = 0f
+    private var top = 0f
+    private var right = 0f
+    private var bottom = 0f
+    private var centerX = 0f
+    private var centerY = 0f
+    private var background = 0
+    private var uncheckColor = 0
+    private var checkedColor = 0
+    private var borderWidth = 0
+    private var checkLineColor = 0
+    private var checkLineWidth = 0
+    private var checkLineLength = 0f
+    private var uncheckCircleColor = 0
+    private var uncheckCircleWidth = 0
+    private var uncheckCircleOffsetX = 0f
+    private var uncheckCircleRadius = 0f
+    private var checkedLineOffsetX = 0f
+    private var checkedLineOffsetY = 0f
+
+    /**
+     * Color for button when it's unchecked
+     */
+    private var uncheckButtonColor = 0
+
+    /**
+     * Color for button when it's checked
+     */
+    private var checkedButtonColor = 0
+    private var buttonMinX = 0f
+    private var buttonMaxX = 0f
+    private lateinit var buttonPaint: Paint
+    private lateinit var paint: Paint
+    private lateinit var viewState: ViewState
+    private lateinit var beforeState: ViewState
+    private lateinit var afterState: ViewState
+    private var animateState = ANIMATE_STATE_NONE
+    private lateinit var valueAnimator: ValueAnimator
+    private val argbEvaluator = ArgbEvaluator()
+    private var isChecked = false
+    private var enableEffect = false
+    private var shadowEffect = false
+    private var showIndicator = false
+    private var isTouchingDown = false
+    private var isUiInited = false
```

Comment:
@shubertm The variable is for checking if the view initialization has been completed, to prevent cases like user tries to toggle the switch too fast while the previous switch has not been completed (in terms of UI update).

Reply:
Got you, but may be just make it more readable like ```isUiInititialized```. I guess not too long :)

Reply:
@shubertm Definitely. I just renamed it to make it better.
Thanks.